tasks/assemble.js

```diff
@@ -719,7 +719,19 @@ module.exports = function(grunt) {
    * @return {String}         The raw, assembled, uncompiled and unprocessed page
    */
   var injectBody = function(layout, body) {
-    return layout.replace(assemble.engine.bodyRegex, function () { return body; });
+    var indentation = getIndentation(layout);
+    var bodyWithIndentation = _.map(body.split('\n'), function (line){
+      return line.replace(/^/, indentation);
+    }).join('\n');
+    return layout.replace(assemble.engine.bodyRegex, function () { return bodyWithIndentation; });
+  };
+
+  var getIndentation = function(layout) {
+    var bodyRegex = assemble.engine.bodyRegex;
+    var bodyRegexWithIndentation = new RegExp(/\s*/.source + bodyRegex.source);
+
+    return layout.match(bodyRegexWithIndentation)[0]
+      .replace(bodyRegex, '');
   };
```

Comment:
`String.match` returns `null` when no match is found, which would result in `null[0]` and would throw an error. maybe do something like `

```js
var res = layout.match(bodyRegexWithIndentation);
if (!res) {
  return layout;
}
return res[0].replace(bodyRegex, '');
```